### ISSUE TITLE
Start using remote.alertmanager from grafana.ini or remove it from grafana.ini

### ISSUE THREAD

**Issue by dragoangel** [grafana/grafana#82414]
**Why is this needed**:

remote.alertmanager was been added half year ago in https://github.com/grafana/grafana/pull/71318 with comment [`NOTE: this configuration options are not used yet`](https://github.com/grafana/grafana/pull/71318/files#diff-3f8a56cf48f018d19dd5378a288ed6b9180d60ef82649c98339f93b3523bd7ceR1192) but it continues to be unused. Administrators sees that section in Grafana settings, but doesn't see it doesn't connected to any functions, leading to missliding information.

**What would you like to be added**:

Start to honor `[remote.alertmanager]` settings in grafana.ini or remove them from configuration

**Who is this feature for?**

To fix misleading settings that doesn't provide any functional changes to Grafana functionality

**Comment by wonderphil** [comment]
be really good if we could start using it

**Comment by ChristianTackeGSI** [comment]
Please unstale. Something like this or #88020 would be appreciated.

**Comment by Starttoaster** [comment]
It would be nice to see movement on this. This is the one feature that stops my team from being able to have completely stateless Grafana deployments in k8s. We have to have a persistent volume just to retain the setting to use our remote Alertmanager for alerting. Everything else comes through the provisioning API.

**Comment by dragoangel** [comment]
> It would be nice to see movement on this. This is the one feature that stops my team from being able to have completely stateless Grafana deployments in k8s. We have to have a persistent volume just to retain the setting to use our remote Alertmanager for alerting. Everything else comes through the provisioning API. 

Hi @Starttoaster I actually have stateless grafana, but I am using alertmanager as notification endpoint, more over - I not filling just one aleranager, but using iteration to fill all replicas via sts service, this way I know that each replica getting alert.

**Comment by Starttoaster** [comment]
How are you getting it to prefer your Prometheus Alertmanagers over the Grafana built-in Alertmanager? That's the thing we're stuck on currently, Grafana won't send alerts to our self-hosted Prometheus Alertmanager statefulset, because it defaults that one to disabled and the built-in Grafana Alertmanger defaults to enabled. We have to disable that Grafana Alertmanager every time the container restarts, or only once with a Persistent Volume (which is not stateless by definition.)

**Comment by dragoangel** [comment]
> How are you getting it to prefer your Prometheus Alertmanagers over the Grafana built-in Alertmanager? That's the thing we're stuck on currently, Grafana won't send alerts to our self-hosted Prometheus Alertmanager statefulset, because it defaults that one to disabled and the built-in Grafana Alertmanger defaults to enabled. We have to disable that Grafana Alertmanager every time the container restarts, or only once with a Persistent Volume (which is not stateless by definition.)

Hi @Starttoaster sorry for a late reply, missed to read email 😅, I using old well working way for me that properly utilize HA Alertmanager via contact points in Grafana. I highlighted in another issue that way in which Grafana propose to utilize external alertmanager is lacking of proper HA utilization, as to do it as alertmanager client need sent alert to all alertmanager replicas, not just one - as alertmanager DO NOT replicate it's alerts between replicas, only silences, this how I achieve it:
```
apiVersion: v1
kind: Secret
metadata:
  namespace: {{ template "kube-prometheus-stack-grafana.namespace" . }}
  name: grafana-{{ template "kube-prometheus-stack.name" $ }}-contact-points
  labels:
    {{- if $.Values.grafana.sidecar.alerts.label }}
    {{ $.Values.grafana.sidecar.alerts.label }}: {{ ternary $.Values.grafana.sidecar.alerts.labelValue "1" (not (empty $.Values.grafana.sidecar.alerts.labelValue)) | quote }}
    {{- end }}
    app: {{ template "kube-prometheus-stack.name" $ }}-grafana
{{ include "kube-prometheus-stack.labels" $ | indent 4 }}
type: Opaque
stringData:
  contact-points.yaml: |-
    apiVersion: 1
    contactPoints:
      - name: prometheus-alertmanager
        receivers:
        {{- if gt ( .Values.alertmanager.alertmanagerSpec.replicas | int ) 1 }}
        {{- range $i, $e := until ( .Values.alertmanager.alertmanagerSpec.replicas | int ) }}
          - uid: prometheus-alertmanager-{{$i}}
            type: prometheus-alertmanager
            settings:
              url: http://alertmanager-monitoring-{{$i}}.alertmanager-operated:9093/
        {{- end }}
        {{- else }}
          - uid: prometheus-alertmanager
            type: prometheus-alertmanager
            settings:
              url: http://alertmanager-monitoring.alertmanager-operated:9093/
        {{- end }}
    deleteContactPoints:
      - uid: '' # default contact point uid
    policies:
      - receiver: prometheus-alertmanager
        group_by: ['...']
        group_wait: 0s
        group_internval: 0s
        repeat_interval: 1m
```
Basically I iterate over replicas count and create 1 contact point with multiple receivers, in my case this is 3 replicas of alertmanager deployed in same stack as Grafana by kube-prometheus-stack 😊, and in case there is only 1 replica - I just create 1 receiver without index*. This requires you to have `alertmanager-operated` headless service for your alertmanager. As we have policy to send all alert to  `prometheus-alertmanager` receiver - every alert from Grafana goes directly there.

P.s. you need enable provisioning sidecar for `alerts` in Grafana to get this thing working.

**Comment by Starttoaster** [comment]
Weird. The rest of it worked, but deleteContactPoints didn't delete the builtin alertmanager contact point, and it appears its still configured to be the default in my Grafana instance.

**Comment by dragoangel** [comment]
> Weird. The rest of it worked, but deleteContactPoints didn't delete the builtin alertmanager contact point, and it appears its still configured to be the default in my Grafana instance.

Well this works for me ilfor 3 years over tons of releases.
